### ISSUE TITLE
`monitoring_role_arn` for `"aws_db_instance" "this_mssql"` resource calculates empty value and generates error in case `monitoring_interval` > 0

### ISSUE THREAD

**Issue by yxycman** [terraform-aws-modules/terraform-aws-rds#242]
Terraform: 0.12.25
AWS_provider: 2.70.0
module version: 2.15.0, 2.17.0

During creation of MSSQL backed RDS instance, i am getting the following error:
```Error: Error modifying DB Instance db-host: InvalidParameterCombination: A MonitoringRoleARN value is required if you specify a MonitoringInterval value other than 0```
as a result of the following interpolation:
```var.monitoring_interval > 0 ? coalesce(var.monitoring_role_arn, aws_iam_role.enhanced_monitoring.*.arn, null) : null```

For the `"aws_db_instance" "this"` resource in the same file, `monitoring_role_arn` is calculated with slightly different interpolation and, if i apply it to resource in question everything works fine
```var.monitoring_interval > 0 ? coalesce(var.monitoring_role_arn, join(", ", aws_iam_role.enhanced_monitoring.*.arn), null) : null```

looks like the cause is the type of the `aws_iam_role.enhanced_monitoring.*.arn` item, which is a list and a behavior of the `coalesce` function in case, more than two arguments provided.

```
> coalesce("", ["atata", "aaaa"], null)

> coalesce("", join(",", ["atata", "aaaa"]), null)
atata,aaaa
> coalesce("", ["atata", "aaaa"])

>
Error: Error in function call

  on <console-input> line 1:
  (source code not available)

Call to function "coalesce" failed: all arguments must have the same type.
```

Calling module with:
```
module "rds" {
  source                              = "terraform-aws-modules/rds/aws"
  version                             = "2.17.0"
  identifier                          = var.identifier
  engine                              = "sqlserver-se"
...
  create_monitoring_role             = true
  monitoring_interval                   = 60
...
}
```

https://github.com/terraform-aws-modules/terraform-aws-rds/blob/v2.17.0/modules/db_instance/main.tf#L151

**Comment by farhan-taib** [comment]
Is there any update on the above, looks like version 2.18.0 still has this issue.

`Error: Error creating DB Instance: InvalidParameterCombination: A MonitoringRoleARN value is required if you specify a MonitoringInterval value other than 0.
[2020-10-05T06:43:47.049Z] 	status code: 400, request id: 8ad86abc-428f-458e-a84f-8d9a0e631094`

**Comment by mpardej** [comment]
Getting the same error. Is there any update?

**Comment by bryantbiggs** [comment]
Closed with #318
